### PR TITLE
Add a CCN value to every method in a class

### PR DESCRIPTION
**Pull request by kudashevs**
I've found your package very useful. I want to thank you for the great package.

During the usage, I realized that there is a thing which I really miss in this package. The possibility to see the Cyclomatic complexity of the most complex method in a class is useful, but I really wanted to see the complexity of every method in a class. So, my proposition is to add this possibility, which I think is quite useful. For now it looks like this.

image

### CODE REVIEW

**Comment by niconoe-** [comment]
Thank you very much for this feature, it's an interesting one to go deeper into the details!

On UI/UX side, I'm afraid this could be too verbose on projects with many classes. Do you think you could handle it via an accordion on the class line? A kind of triange-arrow-right could be added for each class-line and onclick on that arrow or the class name, the accordion would collapse (triangle-arrow rotates to be oriented to bottom) revealing the CCN for every methods of the current clicked class.

The default verbosity would still be the same so that current users won't get lost by too much data displayed, but at least the details by method would become available on demand thanks to your work.

**Comment by niconoe-** [comment]
Thank you very much! This looks nice to me!

@Halleck45 what do you think about?

**Review by niconoe- — APPROVED** [review]

File: templates/html_report/css/style.css [hunk 1]
```diff
@@ -578,7 +578,7 @@ table tr td {
     position:relative;
 }
 .js-sort-table td {
-    padding:.125em .325em .125em 0;
+    vertical-align:baseline;
```
Comment (kudashevs) [inline templates/html_report/css/style.css:581]: This one removes an element jumping which happens when you click on a class name.

**Comment by Mte90** [comment]
Maybe for next 3.0.0 @niconoe-?

**Comment by kudashevs** [comment]
Hello @Halleck45 

Could you please provide any feedback about this PR?

This functionality was requested long before in #444 and I find it very useful.
Do you like the whole idea? Should I refine or improve something in this PR?

**Comment by niconoe-** [comment]
I think that rather having a table component, we should migrate to something more flexible to allow sub-tables. I'll try to take some time to search for reliable components about it.

**Comment by niconoe-** [comment]
Sorry for the waiting about the PR. I had to take a long break recently.

@kudashevs  I reworked the incoming v3 with your work in an adapted way in order to manage both UX/UI and not to interfear too much with logic.

I go further than your proposal by managing to display the accessors even if they are ignored from the cyclomatic complexity calculations, with a flag to explain this to the end-user.
I also managed to open the rows via an accordion style without having too much JS changed, and keeping the possibilty to sort rows by their column values.

Here is a sample of how it's displayed now:
selection

I think this way, it's better, and it opens the way to having other metrics on the methods in the future.

It will be part of the v3 and will probably be released in the next RC.

Feel free to suggest or remark stuff about it 😉

**Comment by kudashevs** [comment]
It looks good. Hope to see this functionality soon in the V3 RC.

**Pull request closed without merging**